lib/elastic_record/as_document.rb

```diff
@@ -16,27 +16,40 @@ def as_partial_update_document
 
       changed_fields.each_with_object({}) do |field, result|
         if field_mapping = mappings[field]
-          result[field] = elastic_search_value field, field_mapping
+          result[field] = value_for_elastic_search field, field_mapping
         end
       end
     end
 
-    def elastic_search_value(field, mapping)
+    def value_for_elastic_search(field, mapping)
       value = try field
       return if value.nil?
 
-      value = case mapping[:type]&.to_sym
-              when :object
-                value.respond_to?(:as_search_document) ? value.as_search_document : value
-              when :nested
-                value.map(&:as_search_document)
-              else
-                value
-              end
+      value =
+        case mapping[:type]&.to_sym
+        when :object
+          value_for_elastic_search_object(value)
+        when :nested
+          value.map { |entry| value_for_elastic_search_object(entry) }
+        when :integer_range, :float_range, :long_range, :double_range, :date_range
+          value_for_elastic_search_range(value)
+        else
+          value
+        end
 
       if value.present? || value == false
         value
       end
     end
+
+    def value_for_elastic_search_object(object)
+      object.respond_to?(:as_search_document) ? object.as_search_document : object
+    end
+
+    def value_for_elastic_search_range(range)
+      gte = range.begin unless range.begin == -Float::INFINITY
+      lte = range.end unless range.end == Float::INFINITY
+      {'gte' => gte, 'lte' => lte}
```

Comment:
If the beginning or end are infinity, then our document value is `nil`.  Is this acceptable for an ES range?  The documentation does not mention anything about nil values.

Reply:
Ruby ranges can be specified in reverse:  

```
record = SpecialFieldsModel.new(meta: ['whatever'], book_length: 250..0)
```

This results in `{"gte" => 250, "lte" => 0}`.  Is that expected?

Reply:
OK, some more clarity on this:

1) If gte/lte is nil, _or one or both keys are omitted entirely_, the document indexes successfully, and the range is unbounded on that key(s).

2) If gte > lte, the document fails to index.

```
PUT /range_index/_doc/3 
{
  "expected_attendees" : { 
		"gte": 20,
		"lte": 10
  }
}
=> 
{
	"error": {
		"root_cause": [
			{
				"type": "mapper_parsing_exception",
				"reason": "failed to parse [expected_attendees]"
			}
		],
		"type": "mapper_parsing_exception",
		"reason": "failed to parse [expected_attendees]",
		"caused_by": {
			"type": "illegal_argument_exception",
			"reason": "min value (20) is greater than max value (10)"
		}
	},
	"status": 400
}
```